### DISCUSSION TITLE
any working linux cmd line for Russia Rostelecom

### DISCUSSION THREAD

**Discussion by edvejak** [hufrea/byedpi#155] [General]
I was using: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -d 1. But it stopped working someday. Is anybody using any working linux cmd line for Russian Rostelecom?

**Comment by cravvler9** [comment]
> I was using: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -d 1. But it stopped working someday. Is anybody using any working linux cmd line for Russian Rostelecom?

-F --split 2 --disorder -1+s 
This options work for me on local isp, but they use Rostelecom lines.

**Comment by unitedrus** [comment]
> > I was using: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -d 1. But it stopped working someday. Is anybody using any working linux cmd line for Russian Rostelecom?
> 
> -F --split 2 --disorder -1+s This options work for me on local isp, but they use Rostelecom lines.

not working =(

**Comment by mapron** [comment]
-F option is unknown on byedpi-14.1. Are you using main branch or something? 
```
 #ifdef TCP_FASTOPEN_CONNECT
    {"tfo ",          0, 0, 'F'},
    #endif
 ```
 I see.

**Comment by stormy23dev** [comment]
> > I was using: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -d 1. But it stopped working someday. Is anybody using any working linux cmd line for Russian Rostelecom?
> 
> -F --split 2 --disorder -1+s This options work for me on local isp, but they use Rostelecom lines.

Also doesn't work (Russia Rostelecom Saint-Petersburg)

**Comment by cravvler9** [comment]
> I was using: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -d 1. But it stopped working someday. Is anybody using any working linux cmd line for Russian Rostelecom?

Try it -F --split 1 --disorder -1

**Comment by cravvler9** [comment]
> Are you using main branch or something?

0.14.1 version from aur
![image](https://github.com/user-attachments/assets/9df120db-da70-4784-87e4-1bac52bacec4)

**Reply by kiav** [reply]
Работает в Ростелеком, Владимир
`ciadpi --ip 127.0.0.1 --tfo --split 1 --disorder -1`

Из byedpi-0.14.1.zip

**Comment by edvejak** [comment]
Correct command line now: 
/usr/local/bin/ciadpi-x86_64 -F --split 1 -i 127.0.0.1 -p 10801 --disorder -1

**Comment by L1fter** [comment]
-F --split 2 --disorder -2
Москва, Ростелеком
Шортсы на трубе подтупливают некоторые и часто в логе отлупы вижу такие:
ciadpi[1057512]: recv: Connection reset by peer
ciadpi[1057512]: state: 2
ciadpi[1057512]: send: Resource temporarily unavailable
Но это пока лучший вариант из тех, что у меня есть
Пробовал такие:
--disorder 1 --auto=torst --tlsrec 1+s
--split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h
--split 1 --oob 1 --disorder 3 --mod-http=h,d --fake -1 --tlsrec 3+h --ip 127.0.0.1
-F --split 1 --disorder -1
На них ютуб либо вообще не грузился, либо только превьюхи, но видео не раздуплилось

**Reply by ALV-036** [reply]
VRN Ростелеком       с такой строкой более менее показывает, иногда замирания, ну хоть что то
/opt/byedpi/ciadpi-x86_64 --drop-sack --tfo --md5sig --split 2 --split 3 --tlsrec 3+s --auto=torst --timeout 1

В основном замирает Firefox когда обращается к Linux proxy byedpi   а вот на андроид планшетах с использованием super proxy клиента при работе через тот же proxy  linux byedpi  всё работает быстро

**Reply by antonnagorniy** [reply]
> VRN Ростелеком с такой строкой более менее показывает, иногда замирания, ну хоть что то /opt/byedpi/ciadpi-x86_64 --drop-sack --tfo --md5sig --split 2 --split 3 --tlsrec 3+s --auto=torst --timeout 1
> 
> В основном замирает Firefox когда обращается к Linux proxy byedpi а вот на андроид планшетах с использованием super proxy клиента при работе через тот же proxy linux byedpi всё работает быстро

Работает, спасибо!

**Reply by vehametra** [reply]
Снова актуально

**Comment by edvejak** [comment]
Correct command line now:
Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64  -i 127.0.0.1 -p 10801 -s1 -q1 -f-1 -S -b661 -As
Ubuntu (Gnome): /usr/local/bin/ciadpi-x86_64  -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As

Thanks to zimbabwe303 #58

**Reply by edvejak** [reply]
Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As  still works. But if video isn't showed try another -b, for example -b897

**Reply by cravvler9** [reply]
> Correct command line now: Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -q1 -f-1 -S -b661 -As Ubuntu (Gnome): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As
> 
> Thanks to zimbabwe303 #58

Thanks, this line works for me
-s1 -f-1 -S -b661 -As

**Comment by edvejak** [comment]
ByeDPI v0.15 is ok. Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As (youtube, discord web version)

**Reply by cravvler9** [reply]
Yea, all working fine, last version ByeDPI from aur.

**Comment by L1fter** [comment]
На текущий момент самый лучший вариант, где нет ожидания прогрузки видео и также работают шортсы:
-o1 -o25+s -T3 -At o--tlsrec 1+s

**Reply by edvejak** [reply]
> На текущий момент самый лучший вариант, где нет ожидания прогрузки видео и также работают шортсы: -o1 -o25+s -T3 -At o--tlsrec 1+s

Thank you. It works like a charm

**Reply by cravvler9** [reply]
> На текущий момент самый лучший вариант, где нет ожидания прогрузки видео и также работают шортсы: -o1 -o25+s -T3 -At o--tlsrec 1+s

Хорошо работает, спасибо

**Reply by edvejak** [reply]
> На текущий момент самый лучший вариант, где нет ожидания прогрузки видео и также работают шортсы: -o1 -o25+s -T3 -At o--tlsrec 1+s

Sadly 'o25+s -T3 -At o--tlsrec 1+s' stopped working for me.  Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As (youtube, discord web version) still works as expected. It's very strange that yt-dlp works on one my pc, but it doesn't work on another one: [download] Got error: The read operation timed out. Retrying (1/10)... . I have Lubuntu installation on both pc

**Reply by edvejak** [reply]
Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As (youtube, discord web version, rutracker)

**Reply by edvejak** [reply]
> Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As (youtube, discord web version, rutracker)

usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801  -o1 -o25+s -T3 -At o--tlsrec 1+s

**Reply by erydit** [reply]
> На текущий момент самый лучший вариант, где нет ожидания прогрузки видео и также работают шортсы: -o1 -o25+s -T3 -At o--tlsrec 1+s

Все, в Татарстане на Ufanet больше не работает этот вариант.

**Reply by antonnagorniy** [reply]
На домашнем Ростелекоме работает, а на корпоративном нет:(

**Reply by L1fter** [reply]
чисто для ютуба сделал списочек с хостами и запускаю так
ciadpi --hosts youtube.hosts -o1 -o25+s -T3 -At o--tlsrec 1+s -p 1070
youtube.com
youtu.be
yt.be
googlevideo.com
ytimg.com
ggpht.com
gvt1.com
youtube-nocookie.com
youtube-ui.l.google.com
youtubeembeddedplayer.googleapis.com
youtube.googleapis.com
youtubei.googleapis.com
yt-video-upload.l.google.com
wide-youtube.l.google.com
С такими настройками только труба и будет открываться. Но можно исключения делать. Например, в foxy-proxy

**Reply by edvejak** [reply]
> > Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As (youtube, discord web version, rutracker)
> 
> usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -o1 -o25+s -T3 -At o--tlsrec 1+s

Still working for me, yt-dlp as well

**Reply by kiav** [reply]
> чисто для ютуба сделал списочек с хостами и запускаю так ciadpi --hosts youtube.hosts -o1 -o25+s -T3 -At o--tlsrec 1+s -p 1070 youtube.com youtu.be yt.be googlevideo.com ytimg.com ggpht.com gvt1.com youtube-nocookie.com youtube-ui.l.google.com youtubeembeddedplayer.googleapis.com youtube.googleapis.com youtubei.googleapis.com yt-video-upload.l.google.com wide-youtube.l.google.com С такими настройками только труба и будет открываться. Но можно исключения делать. Например, в foxy-proxy

Что есть "o--tlsrec" ? Я лишнюю буковку убрал.

Для тех, кто помучился в варианте с systemd:

`BYEDPI_OPTIONS="--ip 127.0.0.1 --hosts ./.config/byedpi.youtube -o1 -o25+s -T3 -At --tlsrec 1+s"`

**Reply by kiav** [reply]
При этом в ~/.config/systemd/user/byedpi.service

[Unit]
Description=ByeDPI
Documentation=https://github.com/hufrea/byedpi
Wants=network-online.target
After=network-online.target nss-lookup.target

[Service]
NoNewPrivileges=yes
StandardOutput=null
StandardError=journal
EnvironmentFile=-/etc/byedpi.conf
EnvironmentFile=-%h/.config/byedpi.conf
ExecStart=%h/bin/ciadpi $BYEDPI_OPTIONS
TimeoutStopSec=5s
PrivateTmp=true
ProtectSystem=full

[Install]
WantedBy=default.target

Я там менял только ExecStart

**Reply by kiav** [reply]
Фсё! Ростелеком, Владимир, не работает в yt-dlp. Масса ошибок при загрузке, файл можно закачивать весь день.

[download] Got error: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:979). Retrying (1/inf)...
[download] Got error: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:979). Retrying (2/inf)...
[download] Got error: SocksHTTPSConnectionPool(host='rr7---sn-gvnuxaxjvh-aome.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (3/inf)...
[download] Got error: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:979). Retrying (4/inf)...
[download] Got error: SocksHTTPSConnectionPool(host='rr7---sn-gvnuxaxjvh-aome.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (5/inf)...
[download] Got error: SocksHTTPSConnectionPool(host='rr7---sn-gvnuxaxjvh-aome.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (6/inf)...
[download] Got error: SocksHTTPSConnectionPool(host='rr7---sn-gvnuxaxjvh-aome.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (7/inf)...
[download] Got error: SocksHTTPSConnectionPool(host='rr7---sn-gvnuxaxjvh-aome.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (8/inf)...

А в браузере давно уже выше 720p не посмотришь.

**Reply by edvejak** [reply]
> > > Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -f-1 -S -b661 -As (youtube, discord web version, rutracker)
> > 
> > 
> > usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -o1 -o25+s -T3 -At --tlsrec 1+s
> 
> Still working for me, yt-dlp as well
0.16.2 is ok

**Reply by kiav** [reply]
> [download] Got error: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:979). 

Очень похоже на проблемы самого Python (если у других с ютубиком и этим yt-dlp проблем нет). В исходниках Python есть и _ssl.c и строковая константа TLSV1_ALERT_PROTOCOL_VERSION (берется из исходников openssl скриптом Tools/ssl/make_ssl_data.py).

Проверял на Python версии 3.12.7 и 3.9.20. С каким openssl их собирали не знаю.

**Reply by kiav** [reply]
> [download] Got error: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:979). 

А нет таких проблем при использовании https прокси в США. Так что дело тут именно в byedpi. Похоже, нужно подбирать другие параметры.

**Comment by edvejak** [comment]
ByeDPI v0.16.6 is ok. 
Lubuntu (Lxqt): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -o1 -o25+s -T3 -At --tlsrec 1+s
Ubuntu (Gnome): /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -s1 -q1 -Y -Ar -s5 -o25000+s -At -f-1 -r1+s -As -s1 -o1+s -s-1 -An -b+500
Rostelecom Far Eastern Division (RFED)
(youtube, discord web version, facebook, rutracker and many other web pages)

**Comment by edvejak** [comment]
ByeDPI v0.17 is ok.
ubuntu: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -o1 -o25+s -T3 -At --tlsrec 1+s
 (youtube only. Many other web pages stopped working)

**Comment by xpashkin** [comment]
ByeDPI 17, -o1 --tlsrec -5+se -T3  -At --tlsrec 1+s
youtube and other sites work.
(--tlsrec -5+se - split ClientHello from end -5. 
o25+s work only with long domain names like rr4---sn-u5uuxaxjvhg0-31oe.googlevideo.com)

**Reply by edvejak** [reply]
> ByeDPI 17, -o1 --tlsrec -5+se -T3 -At --tlsrec 1+s youtube and other sites work. (--tlsrec -5+se - split ClientHello from end -5. o25+s work only with long domain names like rr4---sn-u5uuxaxjvhg0-31oe.googlevideo.com)

Unfortunately, your command does not work for me. Nothing works except YouTube. File sharing services have also become unavailable to me (even through VPN, I can receive links, but I cannot download). Apparently, something has changed in Rostelecom's policy

**Comment by xpashkin** [comment]
> Unfortunately, your command does not work for me. Nothing works except YouTube. File sharing services have also become unavailable to me (even through VPN, I can receive links, but I cannot download). Apparently, something has changed in Rostelecom's policy

Yes, it appears that around June 10th, RKN implemented a whitelist for domain names belonging to foreign hosting companies like Cloudflare, AmazonWS, and other CDNs. If a domain name isn’t on the whitelist, return traffic from it is dropped/terminated.

I solved the problem for myself by creating two lists of domain names.

--hosts /opt/ciadpi/dom_tlsrec.txt  -o1 --tlsrec -5+se -An --hosts /opt/ciadpi/dom_fakesni.txt -f-1 -t7 -n www.google.com 

dom_tlsrec.txt - for ClientHello splitting
googlevideo.com
youtube.com
youtu.be
ytimg.com
ggpht.com

dom_fakesni.txt - for fake SNI packets
lmstudio.ai
transfer.xethub.hf.co
cas-bridge.xethub.hf.co
hf.co
cloudflare-ech.com
a.nel.cloudflare.com
soundcloud.com
soundcloud.cloud

**Reply by edvejak** [reply]
> > Unfortunately, your command does not work for me. Nothing works except YouTube. File sharing services have also become unavailable to me (even through VPN, I can receive links, but I cannot download). Apparently, something has changed in Rostelecom's policy
> 
> Yes, it appears that around June 10th, RKN implemented a whitelist for domain names belonging to foreign hosting companies like Cloudflare, AmazonWS, and other CDNs. If a domain name isn’t on the whitelist, return traffic from it is dropped/terminated.
> 
> I solved the problem for myself by creating two lists of domain names.
> 
> --hosts /opt/ciadpi/dom_tlsrec.txt -o1 --tlsrec -5+se -An --hosts /opt/ciadpi/dom_fakesni.txt -f-1 -t7 -n [www.google.com](http://www.google.com)
> 
> dom_tlsrec.txt - for ClientHello splitting googlevideo.com youtube.com youtu.be ytimg.com ggpht.com
> 
> dom_fakesni.txt - for fake SNI packets lmstudio.ai transfer.xethub.hf.co cas-bridge.xethub.hf.co hf.co cloudflare-ech.com a.nel.cloudflare.com soundcloud.com soundcloud.cloud

Hi. I've tried this line: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -o1 -o25+s -T3 -At --tlsrec 1+s --hosts /home/.config/dom_fakesni.txt -f-1 -t7 -n [www.google.com](http://www.google.com/) with dom_fakesni.txt :
lmstudio.ai
transfer.xethub.hf.co
cas-bridge.xethub.hf.co
hf.co
cloudflare-ech.com
a.nel.cloudflare.com
soundcloud.com
soundcloud.cloud
modsfire.com
sharemods.com

but got error. It doesn't work

**Reply by xpashkin** [reply]
Hi. try only -i 127.0.0.1 -p 10801 --hosts /home/.config/dom_fakesni.txt -f-1 -t7 -n www.google.com

with dom_fakesni.txt:
modsfire.com
sharemods.com
cloudflare-ech.com
cloudflare.com
cloudflarestorage.com

In my previous example, the lists of files work independently; in your case, the list will only function if the website doesn't respond to the initial request.  However, this isn't happening because the website does respond to the request, but then the data stream is interrupted.

**Comment by edvejak** [comment]
ExecStart=/usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 --hosts /home/user/.config/dom_fakesni.txt -f-1 -t7 -n www.google.com


With this command even youtube stopped working. Nothing works at all

**Reply by xpashkin** [reply]
This config is just for testing file downloads from modfire.com, it worked for me. You might want to try different values between 5 and 15 instead of -t7, or add --md5sig.

**Reply by edvejak** [reply]
> This config is just for testing file downloads from modfire.com, it worked for me. You might want to try different values between 5 and 15 instead of -t7, or add --md5sig.

Ok. I still need working configuration

**Comment by edvejak** [comment]
Rostelecom lifted the block on cloudfare today. But it's unclear for how long?

**Comment by sakontwist** [comment]
For YouTube, I have to use a complex setup: for accessing YouTube itself and the redirector - OOB, and for the caches, a packet offset of n bytes is used, followed by a split on SNI+m. It seems this breaks the attempt to determine the start of SNI on TSPU.

```
ciadpi-x86_64 -p 1100 -x 1 --buf-size 16384 -g 128 --auto-mode 1 --cache-ttl 14400 --proto udp --udp-fake 5 --split 25+s --ttl 4  \
--auto none --proto http --mod-http=h,d \
--auto none --ipset /opt/byedpi/google.set --proto tls --oob 23+s --timeout 2 --round 1-3 \
--auto none --proto tls --disorder 3 --split 25+s
```

Overall, it works for most sites with --disorder 3 --split 25+s

**Comment by edvejak** [comment]
ByeDPI v0.17 is ok.
Ubuntu: /usr/local/bin/ciadpi-x86_64 -i 127.0.0.1 -p 10801 -o1 -o25+s -T3 -At --tlsrec 1+s
(youtube, discord web version, facebook, epic, rutracker and many other web pages)
File sharing services is available again due to Rostelecom lifted the block on cloudfare
